test/testframe.h

```diff
@@ -0,0 +1,803 @@
+ *
+ * \see TestShutdown(), TestUsage(), TestParseCmdLine()
+ *
+ */
+H5TEST_DLL herr_t TestInit(const char *ProgName, void (*TestPrivateUsage)(FILE *stream),
+                           int (*TestPrivateParser)(int argc, char *argv[]), int TestProcessID);
+
+/**
+ * --------------------------------------------------------------------------
+ * \ingroup H5TEST
+ *
+ * \brief Shuts down the testing framework
+ *
+ * \return void
+ *
+ * \details TestShutdown() shuts down the testing framework by tearing down
+ *          the internal state needed for running tests and freeing any
+ *          associated memory. TestShutdown() should be called after any
+ *          other function from this testing framework is called, and just
+ *          before any optional library shutdown functions, such as H5close().
+ *
+ * \see TestInit()
+ *
+ */
+H5TEST_DLL void TestShutdown(void);
+
+/**
+ * --------------------------------------------------------------------------
+ * \ingroup H5TEST
+ *
+ * \brief Prints out test program usage help text
+ *
+ * \param[in]  stream Pointer to output stream to direct output to
+ *
+ * \return void
+ *
+ * \details TestUsage() prints out the test program's usage help text to
+ *          the given output stream specified in \p stream. This includes the
+ *          general list of command-line arguments accepted by the test
+ *          program, additional test program-specific usage help text printed
+ *          out by the optional callback specified in TestInit() and a list
+ *          of all the tests and their descriptions, as added by AddTest().
+ *          \p stream may be NULL, in which case stdout is used.
+ *
+ *          <b>Note:</b> when a parallel test calls TestUsage(), the output,
+ *          including additional output from the optional callback specified
+ *          in TestInit(), is only printed from the MPI process with rank
+ *          value 0. Any collective operations should currently be avoided in
+ *          the optional callback if one is provided.
+ *
+ * \see AddTest(), TestInit()
+ *
+ */
+H5TEST_DLL void TestUsage(FILE *stream);
+
+/**
+ * --------------------------------------------------------------------------
+ * \ingroup H5TEST
+ *
+ * \brief Prints out miscellaneous test program information
+ *
+ * \param[in]  stream Pointer to output stream to direct output to
+ *
+ * \return void
+ *
+ * \details TestInfo() prints out miscellaneous information for the test
+ *          program, such as the version of the HDF5 library that the program
+ *          is linked against. \p stream may be NULL, in which case stdout is
+ *          used.
+ *
+ *          <b>Note:</b> when a parallel test calls TestInfo(), the output is
+ *          only printed from the MPI process with rank value 0.
+ *
+ */
+H5TEST_DLL void TestInfo(FILE *stream);
+
+/**
+ * --------------------------------------------------------------------------
+ * \ingroup H5TEST
```

Comment:
Would it make more sense to have the doxygen group for these routines be called something like H5TESTFRAME?

Reply:
No particular opinion on the group name, especially since we don't currently have any testing documentation and this documentation won't get parsed by doxygen yet anyway until we decide to start doing that. H5TEST was just a generic name to keep all testing documentation together, but having a separate group for this framework would work just as well.